presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -414,6 +414,11 @@ public void addFunctionSignatures(IdentityHashMap<FunctionCall, Signature> infos
         return result;
     }
 
+    public IdentityHashMap<Expression, FieldId> getColumnReferenceFields()
+    {
+        return new IdentityHashMap<>(columnReferences);
```

Comment:
There is just one use of `getTypes()` and it doesn't require copying (`unmodifableMap` can be used there), so maybe change that one too.